### ISSUE TITLE
No speed-up in my implementation too

### ISSUE THREAD

**Issue by LSC527** [orobix/fwdgrad#7]
I implemented this papaer with [torch.autograd.forward_ad](https://pytorch.org/tutorials/intermediate/forward_ad_usage.html). However, fwd gradient showed no speed-up compared to fwd+bwd.

**Comment by DavideTr8** [comment]
It would be interesting for us to see your implementation as well. If you want, you can make a PR to our repo with your code. So we can have multiple implementations available.

**Comment by LittleWork123** [comment]
I ran the code from the repository, but I couldn't replicate the results mentioned in the paper, especially regarding the CNN. I used the hyperparameter settings specified in the paper. 

May I inquire if there are alternative parameter settings available?

![image](https://github.com/orobix/fwdgrad/assets/125523878/a59107af-7087-428b-aab8-9f0d81cb1882)

**Comment by DavideTr8** [comment]
Hi, unfortunately we weren't able to reproduce the same results too.
The hyperparameters we used are the same reported in the paper, but we don't now if alternative hyperparameters settings are available.

We believe that the difference between our implementation and the official one are due to the fact that they did not use functorch
